### PR TITLE
Adding support for uncertainty extraction to NDDataArray

### PR DESCRIPTION
**Pull request by bmorris3**
In #81, I added translators for objects in `astropy.nddata`. In this PR, I add support for extracting uncertainties from glue `Data` objects and translating them into the `uncertainty` attr in `NDDataArray` objects. This was missing in the previous PR, but should have been there.

### CODE REVIEW

File: glue_astronomy/translators/nddata.py [hunk 1]
```diff
@@ -70,20 +73,45 @@ def to_object(self, data_or_subset, attribute=None):
 
         data, subset_state = _get_data_and_subset_state(data_or_subset)
 
-        if isinstance(data.coords, WCS) or isinstance(data.coords, BaseHighLevelWCS):
+        if (
+            isinstance(data.coords, WCS) or
+            isinstance(data.coords, BaseHighLevelWCS) or
+            isinstance(data.coords, SpectralCoordinates)
+        ):
```
Comment (pllim) [inline glue_astronomy/translators/nddata.py:80]:
```suggestion
        if isinstance(data.coords, (WCS, BaseHighLevelWCS, SpectralCoordinates)):
```

File: glue_astronomy/translators/nddata.py [hunk 2]
```diff
@@ -70,20 +73,45 @@ def to_object(self, data_or_subset, attribute=None):
...
             raise TypeError('data.coords should be an instance of Coordinates or WCS')
 
+        component_labels = [d.label for d in data.component_ids()]
+        if attribute is None:
+            if 'data' in component_labels:
+                # by default look for the data attribute
+                attribute = 'data'
+            elif 'flux' in component_labels:
+                # if there is no data attribute, try flux
+                attribute = 'flux'
```
Comment (pllim) [inline glue_astronomy/translators/nddata.py:94]:
```suggestion
            for desired_label in ('data', 'flux'):
                if desired_label in component_labels:
                    attribute = desired_label
                    break
```

**Comment by pllim** [comment]
CI failures appear related.

**Review by pllim — APPROVED** [review]

**Comment by dhomeier** [comment]
I see. Would it make sense to have `to_data` convert other types with `represent_as(StdDevUncertainty)` then? That's probably a bit double-sided, as other applications may have use for the original forms, even if jdaviz does not a this point.
But otherwise I think it should check for the correct type on input, as the present implementation would e.g. silently interpret an `InverseVariance` as `StdDevUncertainty` and return it as such.

**Comment by bmorris3** [comment]
Good idea @dhomeier! Implemented in https://github.com/glue-viz/glue-astronomy/pull/86/commits/91d8dbc8989871182d358dbd5fd46ccc7e29cdc2.

File: glue_astronomy/translators/tests/test_nddata.py [hunk 3]
```diff
@@ -136,6 +136,22 @@ def test_from_nddata(cls, kwargs, data_attr):
...
+        data=data,
+        uncertainty=uncertainty
+    )
+    data_collection = DataCollection()
+    data_collection['data'] = spec
+
+    data = data_collection['data']
+    spec_new = data.get_object(NDDataArray)
+    assert isinstance(spec_new.uncertainty, StdDevUncertainty)
+    assert_equal(spec_new.uncertainty.array, uncertainty.array)
```
Comment (dhomeier) [inline glue_astronomy/translators/tests/test_nddata.py:152]:
```suggestion
@pytest.mark.parametrize('uncertainty_type', (StdDevUncertainty, VarianceUncertainty, InverseVariance))
def test_nddata_uncertainty(uncertainty_type):
    data = [[2, 3], [4, 5]] * u.Jy
    uncertainty = uncertainty_type(np.ones_like(data.value))
    spec = NDDataArray(
        data=data,
        uncertainty=uncertainty
    )
    data_collection = DataCollection()
    data_collection['data'] = spec

    data = data_collection['data']
    spec_new = data.get_object(NDDataArray)
    assert isinstance(spec_new.uncertainty, StdDevUncertainty)
    if  isinstance(uncertainty, StdDevUncertainty):
        assert_equal(spec_new.uncertainty.array, uncertainty.array)
    else:
        assert_equal(spec_new.uncertainty.array, uncertainty.represent_as(StdDevUncertainty).array)
```

**Review by dhomeier — CHANGES_REQUESTED** [review]
Great, thank! I'd still add the tests for the other input types.

**Comment by dhomeier** [comment]
Will need imports for the other uncertainties; also we may have to compare to `spec.uncertainty` since that should have the correct units set (not sure if `represent_as` will transform it correctly otherwise).

**Comment by bmorris3** [comment]
@dhomeier Done and passing 💯

**Review by dhomeier — APPROVED** [review]

**Comment by bmorris3** [comment]
@dhomeier Would you consider this a bugfix, for soon release? We will use it downstream in https://github.com/spacetelescope/jdaviz/pull/2039.

**Review by astrofrog — APPROVED** [review]
Looks good to me!

**Pull request merged**